python/inner-sample-molecules-RNN.py

```diff
@@ -3,20 +3,21 @@
 """
 
 import argparse
-import os
 import os.path
 import torch
 from tqdm import tqdm
 
 from datasets import Vocabulary, SelfiesDataset
 from models import RNN
 from functions import read_smiles
+from NPS_generation.functions import set_seed, seed_type
 
 parser = argparse.ArgumentParser()
 
 ## build the CLI
 parser.add_argument('--database', type=str)
 parser.add_argument('--representation', type=str)
+parser.add_argument('--seed', type=seed_type, default=None, nargs="?", help="Random seed")
```

Comment:
All processes that make up `snakemake` steps need to have some way of invoking `set_seed` (it is not sufficient to call this function before running `snakemake` since all steps run in their own independent processes).